### ISSUE TITLE
python orchestration and terraform

### ISSUE THREAD

**Issue by vjagadish1989** [hashicorp/terraform#23351]
Hello Terraform experts,

Maybe this feature already exists - I've found terraform is a great tool for declarative infrastructure as code. In addition to terraform, I have a DAG of several python scripts that need to run as part of the workflow - the scripts are idempotent and it should be safe to re-run them. 

I'm looking for some orchestration to execute (in parallel):
1. a DAG of custom python scripts
2. terraform configuration

what's the recommended best practice for this? 

1. Should each "python" script be modeled as a null resource with a "depends_on" relationship with other resources? 

2. Alternately, I could model each script as a custom resource (with its own provider), but that requires re-implementing a bunch of the script-logic in "goLang"

have folks thought about other ways to integrate python and terraform, while still preserving concurrency?

**Comment by hashibot** [comment]
Hello! :wave: 

This issue doesn't seem to be following one of the standard issue templates available from [the "New Issue" page](https://github.com/hashicorp/terraform/issues/new/choose). The information requested in those templates is important for my human friends to fully understand what's going on.

Please feel free to open a new issue using either the Bug Report or Feature Request template, as appropriate, and then the Terraform maintainers will review it. Thanks!

Please note that we use GitHub issues for tracking bugs and enhancements rather than for questions. While we may be able to help with certain simple problems here it's better to use [the community forum](https://discuss.hashicorp.com/c/terraform-core), where there are more people ready to help. The GitHub issues here are generally monitored only by our few core maintainers.
